pandas/core/series.py

```diff
@@ -879,43 +879,59 @@ def __unicode__(self):
         width, height = get_terminal_size()
         max_rows = (height if get_option("display.max_rows") == 0
                     else get_option("display.max_rows"))
-        if max_rows and len(self.index) > max_rows:
-            result = self._tidy_repr(min(30, max_rows - 4))
-        elif len(self.index) > 0:
-            result = self._get_repr(print_header=True,
-                                    length=len(self) > 50,
-                                    name=True,
-                                    dtype=True)
-        elif self.name is None:
-            result = u('Series([], dtype: %s)') % (self.dtype)
+        # if max_rows and len(self.index) > max_rows:
+        #     print 'here'
+        #     result = self._tidy_repr(min(300, max_rows - 4))
+        # elif len(self.index) > 0:
+        #     result = self._get_repr(print_header=True,
+        #                             length=len(self) > 50,
+        #                             name=True,
+        #                             dtype=True)
+        # elif self.name is None:
+        #     result = u('Series([], dtype: %s)') % (self.dtype)
+        # else:
+        #     result = u('Series([], name: %s, dtype: %s)') % (self.name,
+        #                                                      self.dtype)
+
+        max_rows = get_option("display.max_rows")
+        max_cols = get_option("display.max_columns")
+        show_dimensions = get_option("display.show_dimensions")
+        if get_option("display.expand_frame_repr"):
+            width, _ = fmt.get_console_size()
         else:
-            result = u('Series([], name: %s, dtype: %s)') % (self.name,
-                                                             self.dtype)
-        return result
+            width = None
 
-    def _tidy_repr(self, max_vals=20):
-        """
+        self.to_string(buf=buf, name=self.name, dtype=self.dtype,
+                       max_rows=max_rows, line_width=width,
+                       show_dimensions=show_dimensions)
+        result = buf.getvalue()
 
-        Internal function, should always return unicode string
-        """
-        if max_vals > 1:
-            num = max_vals // 2
-        else:
-            num = 1
-            max_vals = 2
-        head = self.iloc[:num]._get_repr(print_header=True, length=False,
-                                         dtype=False, name=False)
-        tail = self.iloc[-(max_vals - num):]._get_repr(print_header=False,
-                                                       length=False,
-                                                       name=False,
-                                                       dtype=False)
-        result = head + '\n...\n' + tail
-        result = '%s\n%s' % (result, self._repr_footer())
-
-        return compat.text_type(result)
+        return result
+
+    # def _tidy_repr(self, max_vals=20):
+    #     """
+
+    #     Internal function, should always return unicode string
+    #     """
```

Comment:
if this is not needed, pls just remove it (rather than comment out). If you want to keep the commented out for some reason, then put a comment why its here.